app/models/taxon.rb

```diff
@@ -37,6 +38,20 @@ def redirected?
     publication_state == "unpublished" && !redirect_to.nil?
   end
 
+  def draft_previously_published?
+    previous_state, latest_state = last_two_publication_states
+    previous_state && latest_state == "draft"
+  end
+
+  def ordered_publication_state_history
+    ordered_history = state_history.sort_by { |k, _v| k }
+    ordered_history.map!(&:second)
+  end
+
+  def last_two_publication_states
+    ordered_publication_state_history.last(2)
+  end
```

Comment:
The only other place this is called is in the tests? I'd suggest just saying the following in `draft_previously_published?` (`.last(2)` doesn't make any difference apparently).

```ruby
previous_state, latest_state = ordered_publication_state_history
```

Reply:
That would assign the first two states to `previous_state` and `latest_state`.

I could reverse the order in `ordered_publication_state_history` thought

Reply:
I'll put it back in.